src/mailEclipse.php

```diff
@@ -655,15 +655,26 @@ private static function getMissingParams($arg, $params)
          * Determine if a builtin type can be found.
          * Not a string or object as a Mocked::class can work there.
          *
-         * ->getType() needs to be cast to string to get the protected prop.
+         * getName() is undocumented alternative to casting to string.
+         * https://www.php.net/manual/en/class.reflectiontype.php#124658
+         *
+         * @var \ReflectionType $reflection
          */
-        $argType = (string) collect($params)->where('name', $arg)->first()->getType();
+        $reflection = collect($params)->where('name', $arg)->first()->getType();
 
-        $argType = self::TYPES[$argType] ?? null;
+        if (version_compare(phpversion(), '7.1', '>=')) {
```

Comment:
Why is this condition?

Reply:
Hi @4n70w4, are you talking about the version compare?

Reply:
What that is there for is one of the methods are available on php versions below 7.1, the `__toString()` on `ReflectionType`, this was replace by `getName()` on the same class, but not documented, and only available over PHP version 7.1. 

Also `ReflectionNameType` isn't always returned from my testing when making the change, `ReflectionType was the most consistent, I recall when it was a `ReflectionNamedType` it was something that wasn't going to be any point in passing to Mocker.